src/components/counter/counter.test.tsx

```diff
@@ -0,0 +1,29 @@
+import { describe, expect, test } from 'vitest'
+import { render } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { Counter } from './counter'
+import userEvent from '@testing-library/user-event'
+
+const user = userEvent.setup()
+
+describe('Counter Component', () => {
+  test('Test that component renders initial value provided in props', () => {
+    const { getByText } = render(<Counter initialValue={7} />)
+    const renderedValue = getByText('7')
+    expect(renderedValue).toBeInTheDocument()
```

Comment:
`getByText` doesn't return a value, it returns an html element, so your naming seem a little bit off. 
But usually, this assignment is skipped:

```suggestion
    expect(screen.getByText('7')).toBeInTheDocument()
```